nemo/utils/data_utils.py

```diff
@@ -0,0 +1,289 @@
+
+    Args:
+        store_path: a path to an object on an object store
+
+    Returns:
+        Path to the same object in local cache.
+    """
+
+    if store_path.startswith('ais://'):
+        endpoint = ais_endpoint()
+        if endpoint is None:
+            raise RuntimeError(f'AIS endpoint not set, cannot resolve {path}')
+
+        local_ais_cache = os.path.join(ais_cache_base(), ais_endpoint_to_dir(endpoint))
+        store_bucket, store_object = bucket_and_object_from_uri(store_path)
+        local_path = os.path.join(local_ais_cache, store_bucket, store_object)
+    else:
+        raise ValueError(f'Unexpected store path format: {store_path}')
+
+    return local_path
+
+
+def get_object(path: str, force: bool = False, num_retries: int = 5) -> str:
+    """Download an object from a store path and return the local path.
+    If the input `path` is a local path, then nothing will be done, and
+    the original path will be returned.
+
+    Args:
+        path: path to an object
+        force: force download, even if a local file exists
+        num_retries: number of retries if the get command fails
+    
+    Returns:
+        Local path of the object.
+    """
+    if path.startswith('ais://'):
+        endpoint = ais_endpoint()
+        if endpoint is None:
+            raise RuntimeError(f'AIS endpoint not set, cannot resolve {path}')
+
+        local_path = store_path_to_local_path(store_path=path)
+
+        if not os.path.isfile(local_path) or force:
+            # Either we don't have the file in cache or we force download it
+            # Enhancement: if local file is present, check some tag and compare against remote
+            local_dir = os.path.dirname(local_path)
+            if not os.path.isdir(local_dir):
+                os.makedirs(local_dir, exist_ok=True)
+
+            cmd = [ais_binary(), 'get', path, local_path]
+
+            # for now info, later debug
+            logging.debug('Downloading from AIS')
+            logging.debug('\tendpoint    %s', endpoint)
+            logging.debug('\tpath:       %s', path)
+            logging.debug('\tlocal path: %s', local_path)
+            logging.debug('\tcmd:        %s', subprocess.list2cmdline(cmd))
+
+            done = False
+            for n in range(num_retries):
+                if not done:
+                    try:
+                        # Use stdout=subprocess.DEVNULL to prevent showing AIS command on each line
+                        subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+                        done = True
+                    except subprocess.CalledProcessError as err:
+                        logging.warning('Attempt %d of %d failed with: %s', n + 1, num_retries, str(err))
+
+            if not done:
+                raise RuntimeError('Download failed: %s', subprocess.list2cmdline(cmd))
+
+        return local_path
+
+    else:
+        # Assume the file is local
+        return path
+
+
+class StoreObject:
```

Comment:
Make the name explicit by calling it `AIStoreObject` or `AIStoreObjectWeakRef`. Confusing unless the class is more detached from AIStore.

Reply:
Agree with Som on the naming - I'd rename everything from "store" to "aistore", otherwise it's not immediately clear what kind of store we refer to. Or if you want to make API generic so that we can add more things in the future, then maybe "Store" is not a good name and we should rename it to "cluster" or "remote" or something like that?